### ISSUE TITLE
False positive on 'useless_let_if_seq'

### ISSUE THREAD

**Issue by topecongiro** [rust-lang-nursery/rust-clippy#2176]
When I ran `cargo clippy` against rustfmt repository, I got the following suggestion, which is incorrect:
```
warning: `if _ { .. } else { .. }` is an expression
   --> src/visitor.rs:189:9
    |
189 | /         let mut unindent_comment = self.is_if_else_block && !b.stmts.is_empty();
190 | |         if unindent_comment {
191 | |             let end_pos = source!(self, b.span).hi() - brace_compensation - remove_len;
192 | |             let snippet = self.snippet(mk_sp(self.last_pos, end_pos));
193 | |             unindent_comment = snippet.contains("//") || snippet.contains("/*");
194 | |         }
    | |_________^ help: it is more idiomatic to write: `let <mut> unindent_comment = if unindent_comment { ..; snippet.contains("//") || snippet.contains("/*") } else { self.is_if_else_block && !b.stmts.is_empty() };`
    |
    = note: #[warn(useless_let_if_seq)] on by default
    = note: you might not need `mut` at all
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.166/index.html#useless_let_if_seq
```

This looks similar to #975.
I am using 0.0.166.

**Comment by oli-obk** [comment]
Is there an idiomatic way to rewrite this? Or should we just not lint?

**Comment by topecongiro** [comment]
```rust
let mut cond: bool = foo();
if cond {
    // ...;
    cond = bar();
};
```

could be rewritten as

```rust
let cond = if foo() {
    // ...;
   bar()
} else {
    false
};
```

but I am not certain if I can call this idiomatic.

**Comment by oli-obk** [comment]
Well... you can also do

```rust
let cond = foo() || {
    // ...;
    bar()
};
```

**Comment by topecongiro** [comment]
@oli-obk Thanks! And sorry for nitpicking, that changes the semantics of the original code, as we want to execute `bar()` iff `foo()` is true. What we want is:
```rust
let cond = foo() && {
    // ...;
    bar()
};
```
